src/pages/experiments/[experimentId]/plots-and-tables/components/LegendEditor.jsx

```diff
@@ -1,71 +1,79 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { Radio, Form } from 'antd';
+import _ from 'lodash';
+
+const defaultOption = {
+  positions: 'corners',
+  showEnableOption: true,
+  showValueOption: true,
+};
 
 const LegendEditor = (props) => {
   const {
-    onUpdate, legendEnabled, legendPosition, legendOptions,
+    onUpdate, config, option,
   } = props;
 
-  const onChange = (e) => {
-    onUpdate({ legendEnabled: e.target.value });
-  };
-  const changePosition = (value) => {
-    onUpdate({ legendPosition: value.target.value });
+  const positions = {
+    corners: {
+      'top-left': 'Top left',
+      'top-right': 'Top right',
+      'bottom-left': 'Bottom left',
+      'bottom-right': 'Bottom right',
+    },
+    'top-bottom': {
+      top: 'Top',
+      bottom: 'Bottom',
+    },
+    heatmap: {
+      horizontal: 'Horizontal',
+      vertical: 'Vertical',
+    },
   };
-  let position = null;
-  if (legendOptions === 'top-bot') {
-    position = (
-      <>
-        <div> Position</div>
-        <Form.Item>
-          <Radio.Group
-            onChange={(value) => changePosition(value)}
-            value={legendPosition}
-          >
-            <Radio value='top'>Top</Radio>
-            <Radio value='bottom'>Bottom</Radio>
-          </Radio.Group>
-        </Form.Item>
-      </>
-    );
-  }
-  if (legendOptions === 'corners') {
-    position = (
-      <>
-        <div>Position</div>
-        <Form.Item>
-          <Radio.Group
-            onChange={(value) => changePosition(value)}
-            value={legendPosition}
-          >
-            <Radio value='top-left'>Top-Left</Radio>
-            <Radio value='top-right'>Top-Right</Radio>
-            <Radio value='bottom-left'>Bot-Left</Radio>
-            <Radio value='bottom-right'>Bot-Right</Radio>
-          </Radio.Group>
-        </Form.Item>
-      </>
-    );
-  }
+
+  const elOption = _.merge(defaultOption, option);
```

Comment:
check if we still need to use this merge function

Reply:
fixed in new commit